src/main/java/core/basesyntax/shop/strategy/FileWriteStrategy.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.shop.strategy;
+
+import core.basesyntax.shop.service.WriteToFile;
+import core.basesyntax.shop.service.impl.WriteToCsvFile;
+
+public class FileWriteStrategy {
```

Comment:
You may write all this checks in your WriteToCsvFile class if you wish to. So seems you just don't need this class.

Reply:
the same as above

Reply:
Then it's violation of Open/Close principle. You should use separate handlers for any type of file. And seems use strategy pattern. But it's also you to decide.